### DISCUSSION TITLE
Initial Build Failing

### DISCUSSION THREAD

**Discussion by CapitaOrb** [mavlink-router/mavlink-router#380] [Q&A]
I have setup mavlink-router several times over the last 3 months on some Raspberry Pi 4s and Zero 2Ws - always fresh installs. I've never ran into a problem when building, however the build keeps freezing at 18/24 Linking target examples/px4-offboard-mode.

It has frozen at this exact step 5 times now and I have wiped the SD card between each install/build attempt.
The Pi still responds to enter key presses at an 8 second delay althought the build is frozen.
Any ideas?

![image](https://user-images.githubusercontent.com/106403270/207635561-06a4e47a-355a-4b1e-949e-e63f3f317ae5.png)

![image](https://user-images.githubusercontent.com/106403270/207635279-8c4df0d0-2c5e-41d2-932c-ff1abfa1b477.png)

UPDATE: After installing the packages that were missing in the above images (gtest and clang-tidy), the build halts at "15/46 Compiling C++ object src/mavlink-routerd.p/ulog.cpp.o"

![image](https://user-images.githubusercontent.com/106403270/207651360-d314ea32-a928-4415-ba3c-10a69a41796e.png)

**Comment by jbeyerstedt** [comment]
Compiling on an RPi Zero usually takes quite some time as it's only a single core processor and has quite low RAM. It might help to reduce the number of compiler threads in the build command, e.g. `sudo ninja -j1 -C build` to just run one compiler command at a time.

From what I remember, some steps just take a long time, especially if your RPi is not well cooled. You just need to be patient.

**Reply by CapitaOrb** [reply]
Will do thank you
